src/Gui/MainWindow.cpp

```diff
@@ -2222,6 +2230,121 @@ void MainWindow::showStatus(int type, const QString& message)
+        constexpr int radius = 2;
+        constexpr int iconTotalHeight = 24;
+        constexpr int iconSymbolHeight = iconTotalHeight - 2 * margin;
+
+        const QFont font(QStringLiteral("sans"), 10, QFont::Bold);
+        const QFontMetrics fm(font);
+        const QString text = QString::fromUtf8(inputRepresentation(key));
+        const QRect textBoundingRect = fm.tightBoundingRect(text);
+
+        const unsigned symbolWidth =
+            std::max(textBoundingRect.width() + padding * 2, iconSymbolHeight);
+
+        const QRect keyRect(margin, margin, symbolWidth, 18);
+
+        QPixmap pixmap(symbolWidth + margin * 2, iconTotalHeight);
+        pixmap.fill(Qt::transparent);
+
+        QPainter painter(&pixmap);
+        painter.setRenderHint(QPainter::Antialiasing);
+        painter.setPen(QPen(color, 2));
+        painter.setFont(font);
+        painter.drawRoundedRect(keyRect, radius, radius);
+        painter.drawText(
+            // adjust the rectangle so it is visually centered
+            // this is important for characters that are below baseline
+            keyRect.translated(0, -(textBoundingRect.y() + textBoundingRect.height()) / 2),
+            Qt::AlignHCenter,
+            text);
+
+        return pixmap;
+    };
+
+    const auto getKeyImage = [this](InputHint::UserInput key) {
+        const auto& factory = BitmapFactory();
+
+        QPixmap image = [&] {
+            QColor color = d->hintLabel->palette().text().color();
+
+            if (auto iconPath = getCustomIconPath(key)) {
+                return factory.pixmapFromSvg(*iconPath,
+                                             QSize(24, 24),
+                                             {{0xFFFFFF, color.rgb() & RGB_MASK}});
+            }
+
+            return generateKeyIcon(key, color);
+        }();
+
+
+        QBuffer buffer;
+        image.save(&buffer, "png");
+
+        return QStringLiteral("<img src=\"data:image/png;base64,%1\" width=%2 height=24 />")
+            .arg(QLatin1String(buffer.data().toBase64()))
+            .arg(image.width());
+    };
+
+    const auto getHintHTML = [&](const InputHint& hint) {
+        QString message = QStringLiteral("<td valign=bottom>%1</td>").arg(tr(hint.message.c_str()));
+
+        for (const auto& sequence : hint.sequences) {
+            QList<QString> keyImages;
+
+            for (const auto key : sequence.keys) {
+                keyImages.append(getKeyImage(key));
+            }
+
+            message = message.arg(keyImages.join(QString {}));
+        }
+
+        return message;
+    };
+
+    QStringList messages;
+    for (const auto& hint : hints) {
+        messages.append(getHintHTML(hint));
+    }
+
+    QString html = QStringLiteral("<table style=\"line-height: 28px\" height=28>"
+                                  "<tr>%1</tr>"
+                                  "</table>");
+
+    d->hintLabel->setText(html.arg(messages.join(QStringLiteral("<td width=10></td>"))));
```

Comment:
```suggestion
    QString html = QStringLiteral("<table style=\"line-height: 28px\" height=28>"
                                  "<tr>%1</tr>"
                                  "</table>")
                                  .arg(messages.join(QStringLiteral("<td width=10></td>")));

    d->hintLabel->setText(html);
```

Reply:
clang format insists on this formatting:

![image](https://github.com/user-attachments/assets/3fb828fc-e19e-4e1a-abd1-d45782543b59)

I don't find it more readable.